go-controller/pkg/ovn/secondary_layer3_network_controller.go

```diff
@@ -433,14 +403,11 @@ func (oc *SecondaryLayer3NetworkController) addUpdateNodeEvent(node *kapi.Node,
 }
 
 func (oc *SecondaryLayer3NetworkController) addNode(node *kapi.Node) ([]*net.IPNet, error) {
-	hostSubnets, err := oc.allocateNodeSubnets(node, oc.masterSubnetAllocator)
-	if err != nil {
-		return nil, err
-	}
-
-	hostSubnetsMap := map[string][]*net.IPNet{oc.GetNetworkName(): hostSubnets}
-	err = oc.UpdateNodeHostSubnetAnnotationWithRetry(node.Name, hostSubnetsMap, nil)
-	if err != nil {
+	// Node subnet for the secondary layer3 network is allocated by cluster manager.
+	// Make sure that the node is allocated with the subnet before proceeding
+	// to create OVN Northbound resources.
+	hostSubnets, err := util.ParseNodeHostSubnetAnnotation(node, oc.GetNetworkName())
+	if err != nil || len(hostSubnets) < 1 {
```

Comment:
if ` || len(hostSubnets) < 1` , then `err == nil` and there will be no specific error log and no retry. So I think we need to `fmt.Errorf`

Reply:
Ack.